quarkchain/cluster/jsonrpc.py

```diff
@@ -1428,5 +1420,44 @@ def shutdown(self):
         return "pong"
 
     @public_methods.add
-    async def echo(self, params):
-        return params
+    async def qkc_subscribe(self, sub_type, full_shard_id, context):
+        if context is None or full_shard_id is None:
+            return None
+        self.sub_id += 1
+        response = {"jsonrpc": "2.0", "result": self.sub_id, "id": 1}  # TODO: change id
+        await context.send(str(response))
+
+        full_shard_id = shard_id_decoder(full_shard_id)
+        branch = Branch(full_shard_id)
+        shard = self.slave.shards.get(branch, None)
+        if not shard:
+            return None
+
+        if sub_type == "newHeads":
+            await self.fetch_new_head(self.sub_id, shard, context)
+        else:
+            print("other types of subscription")
+            self.subscribers[sub_type].append(self.sub_id)
+
+    async def fetch_new_head(self, sub_id, shard, websocket):
+        latest_header = None
+        while True:
+            header = shard.state.header_tip
+            if not latest_header or header.height != latest_header.height:
+                latest_header = header
+                response = self.response_transcoder(
+                    minor_block_header_encoder(header), sub_id
+                )
+                await websocket.send(str(response))
+
+            await asyncio.sleep(0.5)
+
+    @staticmethod
+    def response_transcoder(result, sub_id):
+
+        return {
+            "jsonrpc": "2.0",
+            "method": "eth_subscription",
```

Comment:
1. `eth_subscription` -> `subscription`
1. `{**result}` -> `result`

also, the format seems should be 
```
{
  "jsonrpc": xxx
  "method": yyy,
  "params": {
     "subscription": sub_id,
     "result": result
   }
}
```